commands/cluster-shards.md

```diff
@@ -0,0 +1,150 @@
+`CLUSTER SHARDS` returns details about the shards of the cluster.
+A shard is defined as a collection of nodes that serve the same set of slots and that replicate from each other.
+A shard may only have a single master at a given time, but may have multiple replicas.
+It is possible for a shard to not be serving any slots while still having replicas.
+
+The command is suitable to be used by Redis Cluster client libraries in order to understand the topology of the cluster.
+A client should issue this command on startup in order to retrieve the map associating cluster *hash slots* with actual node information.
+This map should be used to direct commands to the node that is likely serving the slot associated with a given command.
+In the event the command is sent to the wrong node, this command can then be used to update the topology of the cluster.
+
+The command returns an array of shards, with each shard containing two fields, 'slots' and 'nodes'. 
+
+The 'slots' field is a list of slot ranges served by this shard, stored as pair of integers storing the start and end slots of the ranges.
+For example, if a node owns the slots 1, 2, 3, 5, 7, 8, 9, the slots field would store the ranges like:
+
+```
+1) 1) "slots"
+   2) 1) (integer) 1
+      2) (integer) 3
+      3) (integer) 5
+      4) (integer) 5
+      5) (integer) 7
+      6) (integer) 9
+```
+
+The 'nodes' field contains a list of all nodes within the shard.
+Each individual node is a map of attributes that describe the node. 
```

Comment:
It's slightly confusing that it's described as a map, but the examples are in RESP2. Some mention of how RESP3 maps fall back to RESP2 could be useful here, possibly with a link to the protocol page.

Reply:
Should we consider the pros/cons of using the native RESP 3 map too? I am assuming it is a strategic goal to increase the RESP 3 adoption over time. If backcompat is a concern, one option is to enable the native map only for clients understanding RESP 3 and fallback to RESP 2 otherwise.

Reply:
We use the terminology fairly often, I don't think we should document everywhere that a map looks different between RESP 3 and RESP 2. (Like https://github.com/redis/redis-doc/blob/80a55ab6487f4096e200e3871c083d8468bfe7b3/commands/function-stats.md#L7)

Reply:
If you say so. The examples make it clear anyway.